MobileCenter/MobileCenter/Internals/Channel/MSChannelDefault.m

```diff
@@ -314,13 +326,47 @@ - (void)suspend {
 }
 
 - (void)resume {
-  if (self.suspended && self.enabled) {
+  if (!self.sender.suspended && self.suspended && self.enabled) {
     MSLogDebug([MSMobileCenter logTag], @"Resume channel for group Id %@.", self.configuration.groupId);
     self.suspended = NO;
+    self.doneFlushingCompletion = kMSEmptyDoneFlushingCompletion;
     [self flushQueue];
   }
 }
 
+- (void)notifyWhenDoneFlushingWithCompletion:(MSDoneFlushingCompletionBlock)completion
+{
+  __weak typeof(self) weakSelf = self;
+  dispatch_async(self.logsDispatchQueue, ^{
+    typeof(self) strongSelf = weakSelf;
+    
+    // Decorate the block to execute.
+    strongSelf.doneFlushingCompletion = ^(){
+      typeof(self) toughSelf = weakSelf;
```

Comment:
This is great! ^^
How about superSoftSelf? ;)

Reply:
:) I had to come up with something